engine/installation/linux/ubuntulinux.md

```diff
@@ -116,365 +117,430 @@ packages from the Docker repository:
+>**Warning**: The `docker` group is equivalent to the `root` user. For details
+on how this impacts security in your system, see [*Docker Daemon Attack
+Surface*](../../security/security.md#docker-daemon-attack-surface) for details.
 
 To create the `docker` group and add your user:
 
-1. Log into Ubuntu as a user with `sudo` privileges.
+1.  Log into Ubuntu as a user with `sudo` privileges.
 
-2. Create the `docker` group.
+2.  Create the `docker` group.
+    ```bash
+    $ sudo groupadd docker
+    ```
 
-        $ sudo groupadd docker
+3.  Add your user to `docker` group.
 
-3. Add your user to `docker` group.
+    ```bash
+    $ sudo usermod -aG docker $USER
+    ```
 
-        $ sudo usermod -aG docker $USER
+4.  Log out and log back in so that your group membership is re-evaluated.
 
-4. Log out and log back in.
+5.  Verify that you can `docker` commands without `sudo`.
 
-    This ensures your user is running with the correct permissions.
+    ```bash
+    $ docker run hello-world
+    ```
 
-5. Verify your work by running `docker` without `sudo`.
+	  If this fails, you will see an error:
 
-        $ docker run hello-world
+    ```none
+		Cannot connect to the Docker daemon. Is 'docker daemon' running on this host?
+    ```
 
-	If this fails with a message similar to this:
+	  Check that the `DOCKER_HOST` environment variable is not set for your shell.
+	  If it is, unset it.
 
-		Cannot connect to the Docker daemon. Is 'docker daemon' running on this host?
+    ```bash
+    $ unset DOCKER_HOST
+    ```
 
-	Check that the `DOCKER_HOST` environment variable is not set for your shell.
-	If it is, unset it.
+    You may need to edit your environment in files such as `~/.bashrc` or
+    `~/.profile` to prevent the `DOCKER_HOST` variable from being set
+    erroneously.
 
-### Adjust memory and swap accounting
+### Enable memory and swap accounting
 
-When users run Docker, they may see these messages when working with an image:
+You may see messages similar to the following when working with an image:
 
-    WARNING: Your kernel does not support cgroup swap limit. WARNING: Your
-    kernel does not support swap limit capabilities. Limitation discarded.
+```none
+WARNING: Your kernel does not support cgroup swap limit. WARNING: Your
+kernel does not support swap limit capabilities. Limitation discarded.
+```
 
-To prevent these messages, enable memory and swap accounting on your
-system.  Enabling memory and swap accounting does induce both a memory
-overhead and a performance degradation even when Docker is not in
-use. The memory overhead is about 1% of the total available
-memory. The performance degradation is roughly 10%.
+If you don't care about these capabilities, you can ignore the warning. You can
+enable these capabilities in your kernel by following these instructions. Memory
+and swap accounting incur about a 1% memory overhead and a 10% overall
+performance degradation.
```

Comment:
In changing this paragraph you lose some information that I think was valuable:

- "even when Docker is not in use"
- "of the total available memory"